### PR TITLE
Upgrade akka and akka-http

### PR DESCRIPTION
**Pull request by cocreature**
Was chasing an issue somewhere and thought this might affect it in
some way. It didn’t but I might as well turn the upgrade into a PR.

changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Server.scala [hunk 1]
```diff
@@ -105,21 +103,25 @@ class Server(
...
   }
 
-  private def triggerRunnerName(triggerInstance: UUID): String =
-    triggerInstance.toString ++ "-monitor"
-
-  private def getRunner(triggerInstance: UUID): Option[ActorRef[TriggerRunner.Message]] =
-    ctx
-      .child(triggerRunnerName(triggerInstance))
-      .asInstanceOf[Option[ActorRef[TriggerRunner.Message]]]
+  private def getRunner(triggerInstance: UUID): Future[Option[ActorRef[TriggerRunner.Message]]] = {
```
Comment (cocreature) [inline triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Server.scala:120]: `ctx.child` and `ctx.spawn` are not thread safe and need to be accessed directly from a behavior. afaict, this was always an issue but newer versions of akka throw a runtime error here.

File: ledger-service/http-json-testing/src/main/scala/com/daml/http/WebsocketTestFixture.scala [hunk 2]
```diff
@@ -176,25 +176,24 @@ private[http] object WebsocketTestFixture extends StrictLogging with Assertions
       jwt: Jwt,
       serviceUri: Uri,
       query: String,
-      offset: Option[domain.Offset] = None,
-      keepOpen: Boolean = false)(implicit asys: ActorSystem): Source[Message, NotUsed] =
-    singleClientWSStream(jwt, "query", serviceUri, query, offset, keepOpen)
+      offset: Option[domain.Offset] = None)(implicit asys: ActorSystem): Source[Message, NotUsed] =
```
Comment (cocreature) [inline ledger-service/http-json-testing/src/main/scala/com/daml/http/WebsocketTestFixture.scala:179]: akka-http cancels the output section of the stream if the input is closed which is why we need to keep it open. So far, we got away with not doing that in a lot of places but now you get an exception if you hit this, so we just always keep it open and terminate the output via killswitches or `take`.

File: ledger-service/http-json/src/it/scala/http/WebsocketServiceIntegrationTest.scala [hunk 3]
```diff
@@ -241,17 +249,6 @@ class WebsocketServiceIntegrationTest
       errorResponse.errors should have size 1
   }
 
-  // NB SC #3936: the WS connection below terminates at an appropriate time for
```
Comment (cocreature) [inline ledger-service/http-json/src/it/scala/http/WebsocketServiceIntegrationTest.scala:244]: This was a side effect of not keeping the input open which we now always do.

**Review by stefanobaghino-da — APPROVED** [review]

**Review by aherrmann-da — APPROVED** [review]
LGTM

**Pull request merged**